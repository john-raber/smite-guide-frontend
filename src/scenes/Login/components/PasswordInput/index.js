import React from "react";
import { FormGroup, Label, Input } from "reactstrap";

const PasswordInput = ({ value, onChange }) => {
  return (
    <FormGroup className="PasswordInput">
      <Label for="password">Password</Label>
      <Input
        name="password"
        type="password"
        value={value}
        onChange={e => onChange(e.target.name, e.target.value)}
      />
    </FormGroup>
  );
};

export default PasswordInput;
