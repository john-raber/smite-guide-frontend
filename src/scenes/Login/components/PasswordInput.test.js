import React from "react";
import { shallow } from "enzyme";

import PasswordInput from "./PasswordInput/index";

describe("PasswordInput", () => {
  it("renders correctly", () => {
    let wrapper = shallow(<PasswordInput />);

    expect(wrapper.find(".PasswordInput").length).toEqual(1);
  });

  it("matches its snapshot", () => {
    let wrapper = shallow(<PasswordInput />);

    expect(wrapper).toMatchSnapshot();
  });
});
