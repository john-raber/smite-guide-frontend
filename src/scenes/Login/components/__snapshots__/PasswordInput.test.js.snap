// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordInput matches its snapshot 1`] = `
<FormGroup
  className="PasswordInput"
  tag="div"
>
  <Label
    for="password"
    tag="label"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
      ]
    }
  >
    Password
  </Label>
  <Input
    name="password"
    onChange={[Function]}
    type="password"
  />
</FormGroup>
`;
